### PR TITLE
Add server certificate validation to wazuh agent (Linux and Windows)

### PR DESCRIPTION
**Pull request by Nicogp**
|Related issue|
|---|
|#389|

## Description

This PR adds server certificate validation in the Windows and Linux agent.

Note: verification for macOS agent will be added in a separate PR.

For server connections when the agent is already registered, a new configuration option is added to control which mode will be used:
```
agent:
  verification_mode: full
```

Possible Values:

- full (default):
  - Validates that the server certificate is signed by a trusted CA.
  - Ensures the server hostname matches the certificate's SAN or CN.
- certificate:
  - Validates that the server certificate is signed by a trusted CA.
  - Does not validate the server hostname.
- none:
  - Disables all certificate validation.
  - No checks are performed on the certificate's CA signature or the server hostname.
  - Note: This mode disables critical SSL/TLS security features and is not recommended for production environments.

To perform the registration of the agent, a new option was added to the CLI “--verification-mode”, the values that can take are the same as for the previous case:

`wazuh-agent --register-agent --user user --password pass --url https://serverIP:55000 --verification-mode certificate`

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X

### CODE REVIEW

**Review by aritosteles — APPROVED** [review]
LGTM

**Review by jr0me — APPROVED** [review]
LGTM

File: src/agent/communicator/include/http_request_params.hpp [hunk 1]
```diff
@@ -32,6 +33,7 @@ namespace http_client
                           const std::string& serverUrl,
                           std::string endpoint,
                           std::string userAgent,
+                          std::string verificationMode,
```
Comment (TomasTurina) [inline src/agent/communicator/include/http_request_params.hpp:36]: Add to list of parameters.

File: src/agent/communicator/tests/http_client_test.cpp
Comment (TomasTurina) [inline src/agent/communicator/tests/http_client_test.cpp:1]: Add tests for `certificate` and `none` cases.

File: src/agent/tests/agent_registration_test.cpp
Comment (TomasTurina) [inline src/agent/tests/agent_registration_test.cpp:1]: [Same](https://github.com/wazuh/wazuh-agent/pull/444/files#r1892644772).

File: src/agent/communicator/include/http_request_params.hpp (hunk 1 above)
Comment (Nicogp) [reply src/agent/communicator/include/http_request_params.hpp:37]: Done!

File: src/agent/communicator/tests/http_client_test.cpp
Comment (Nicogp) [reply src/agent/communicator/tests/http_client_test.cpp:1]: Done!

File: src/agent/src/process_options.cpp [hunk 2]
```diff
@@ -30,7 +31,7 @@ void RegisterAgent(const std::string& url,
             agent_registration::AgentRegistration reg(url, user, password, key, name, dbFolderPath);
 
             http_client::HttpClient httpClient;
-            if (reg.Register(httpClient))
+            if (reg.Register(httpClient, verificationMode))
```
[the remark this answers is not in the document]
Comment (Nicogp) [reply src/agent/src/process_options.cpp]: Done!

File: src/agent/tests/agent_registration_test.cpp
Comment (Nicogp) [reply src/agent/tests/agent_registration_test.cpp:1]: Done!

Comment (Nicogp) [reply src/agent/tests/agent_registration_test.cpp:1]: The link seems to be broken

**Review by TomasTurina — APPROVED** [review]
